src/classes/UTIL_AbstractCommon_BATCH.cls

```diff
@@ -0,0 +1,519 @@
+*
+* Additional Where Clause can be configured by passing to constructor
+*
+* If Additional Where Clause is added, it can be validated by executing validateQuery().  If
+* it returns null, it is syntactically correct.  If it returns an Exception object, something about the
+* query is invalid.  The Additional Where and/or Where clause will be applied to the inner query for
+* each batch, if executing in LDV Chunking Mode, and the main batch query if executing in non LDV
+* Chunking mode.
+*
+* Future: Add exception handling, and logging support so implementing clases can inherit.
+* Recommend registering a default handler using an interface that can be switched out/overriden
+* as desired for specific use cases.
+*/
+public abstract with sharing class UTIL_AbstractCommon_BATCH implements Database.Batchable<SObject>, Database.Stateful {
+
+    /**
+     * @description The Maximum LDV Chunk size that is allowed to be used.  If > MAX will use the MAX
+     */
+    private static final Integer MAX_LDV_CHUNK_SIZE = 10000000;
+    /**
+     * @description Default size to use when chunking records
+     */
+    private static final Integer DEFAULT_LDV_CHUNK_SIZE = 1000000;
+    /**
+     * @description Default Maximum Size in # of records between min and max Id before it must run in LDV Chunking Mode
+     */
+    private static final Integer DEFAULT_MAX_NON_LDV_SIZE = 1000000;
+    /**
+     * @description Default size of the batch when re-executing in chunk mode, if not provided
+     */
+    private static final Integer DEFAULT_BATCH_SIZE = 500;
+
+    private Id startId;
+    private Id endId;
+    private Id currentEndId;
+    private Boolean endReached;
+    private Integer batchSize = DEFAULT_BATCH_SIZE;
+    private Integer chunkIteration;
+    private Boolean ldvMode = null;
+    private String additionalWhereClause;
+
+    /**
+     * @description No argument constructor that assumes default batch size of 500, and that ldvMode will be
+     * determined by record distance between min and max Id for main object of batch
+     */
+    public UTIL_AbstractCommon_BATCH() {
+        // batchSize, ldvMode, additionalWhereClause
+        this(DEFAULT_BATCH_SIZE, null,'');
+    }
+
+    /**
+     * @description Constructor that assumes  that ldvMode will be
+     * determined by record distance between min and max Id for main object of batch
+     * @param batchSize Integer Size of each batch
+     */
+    public UTIL_AbstractCommon_BATCH(Integer batchSize) {
+        // batchSize, ldvMode, additionalWhereClause
+        this(batchSize, false, '');
+    }
+
+    /**
+     * @description Constructor that assumes  that ldvMode will be
+     * determined by record distance between min and max Id for main object of batch
+     * @param additionalWhereClause if present, will be applied to the inner query, or the main query if executing in non ldv mode.  If a main where clause is also present both will be encapsulated in () and anded.
```

Comment:
@force2b - I can do that, but typically don't with @param because I read (like 4 years ago) ApexDoc required params to be on one line.  If we don't care, though I am good with that!